autogpt_platform/backend/backend/data/queue.py

```diff
@@ -89,19 +92,31 @@ async def connection(self) -> redis.AsyncRedis:
         return await redis.get_redis_async()
 
     async def publish_event(self, event: M, channel_key: str):
-        message, channel_name = self._serialize_message(event, channel_key)
+        message, full_channel_name = self._serialize_message(event, channel_key)
         connection = await self.connection
-        await connection.publish(channel_name, message)
+        await connection.publish(full_channel_name, message)
 
     async def listen_events(self, channel_key: str) -> AsyncGenerator[M, None]:
-        pubsub, channel_name = self._subscribe(await self.connection, channel_key)
+        pubsub, full_channel_name = self._get_pubsub_channel(
+            await self.connection, channel_key
+        )
         assert isinstance(pubsub, AsyncPubSub)
 
         if "*" in channel_key:
-            await pubsub.psubscribe(channel_name)
+            await pubsub.psubscribe(full_channel_name)
         else:
-            await pubsub.subscribe(channel_name)
+            await pubsub.subscribe(full_channel_name)
 
         async for message in pubsub.listen():
             if event := self._deserialize_message(message, channel_key):
                 yield event
+
+    async def wait_for_event(
+        self, channel_key: str, timeout: Optional[float] = None
+    ) -> M | None:
+        try:
+            return await asyncio.wait_for(
+                anext(aiter(self.listen_events(channel_key))), timeout
+            )
+        except TimeoutError:
+            return None
```

Comment:
I didn't add this on the synchronous `RedisEventBus` because there is no good way to do it in a synchronous context without using threads, and currently it's only used in the async one anyway.